frontend/src/core/codemirror/language/markdown.ts

```diff
@@ -38,13 +38,16 @@ const regexes = pairs.map(
     ] as const,
 );
 
+type QuoteType = '"' | '"""'; // Define a type that restricts to either single or triple quotes
```

Comment:
i think `'` and `'''` are valid as well

Reply:
if its not too much trouble, could we add some tests to `markdown.test.ts`?

```python
  describe("should not modify original code when going in and out", () => {
    const cases = [
      'mo.md("""# Markdown Title\n\nSome content here.""")',
      'mo.md("Some *markdown* content")',
      "mo.md('''# Another Title\nContent''')",
      'mo.md(f"""This is some \\"""content\\"""!""")',
      'mo.md("""This is some \\"""content\\"""!""")',
      'mo.md("This is some \\"content\\"!")',
      "mo.md('This is some \\'content\\'!')",
      'mo.md("""Markdown with an escaped \\"quote\\"""")',
      'mo.md("""# Title\n\n```python\nprint("Hello, Markdown!")\n```""")',
      'mo.md("")',
      'mo.md("# Title without proper delimiters")',
      'mo.md("""   \n# Title\nContent\n   """)',
    ];
    it.each(cases)("should not modify %s", (pythonCode) => {
      const [innerCode] = adapter.transformIn(pythonCode);
      const [wrappedCode] = adapter.transformOut(innerCode);
      expect(pythonCode).toBe(wrappedCode);
    });
  });
```

there is also a case where we cannot keep the `QuoteType`. like when we go from single-line to multi-line, we need to upgrade from single-quote to multi-quote. we should make sure that is handled

Reply:
basically - i don't think we want to downgrade from a `"""` to `"` it if was previously a `"""`. but we do want to upgrade to `"""`, if the markdown becomes multi-line

Reply:
> basically - i don't think we want to downgrade from a `"""` to `"` it if was previously a `"""`. but we do want to upgrade to `"""`, if the markdown becomes multi-line

yeah, this is what i intend to do

Reply:
> i think `'` and `'''` are valid as well

i should've considered these two cases

Reply:
> if its not too much trouble, could we add some tests to `markdown.test.ts`?
> 
> ```python
>   describe("should not modify original code when going in and out", () => {
>     const cases = [
>       'mo.md("""# Markdown Title\n\nSome content here.""")',
>       'mo.md("Some *markdown* content")',
>       "mo.md('''# Another Title\nContent''')",
>       'mo.md(f"""This is some \\"""content\\"""!""")',
>       'mo.md("""This is some \\"""content\\"""!""")',
>       'mo.md("This is some \\"content\\"!")',
>       "mo.md('This is some \\'content\\'!')",
>       'mo.md("""Markdown with an escaped \\"quote\\"""")',
>       'mo.md("""# Title\n\n```python\nprint("Hello, Markdown!")\n```""")',
>       'mo.md("")',
>       'mo.md("# Title without proper delimiters")',
>       'mo.md("""   \n# Title\nContent\n   """)',
>     ];
>     it.each(cases)("should not modify %s", (pythonCode) => {
>       const [innerCode] = adapter.transformIn(pythonCode);
>       const [wrappedCode] = adapter.transformOut(innerCode);
>       expect(pythonCode).toBe(wrappedCode);
>     });
>   });
> ```
> 
> there is also a case where we cannot keep the `QuoteType`. like when we go from single-line to multi-line, we need to upgrade from single-quote to multi-quote. we should make sure that is handled

yeah, i'll add more tests